### PR TITLE
Feat/improve output

### PR DESCRIPTION
**Pull request by carpasse**

### CODE REVIEW

File: src/TapReporter.js [hunk 1]
```diff
@@ -1,60 +1,85 @@
...
+    this.logger.info('\n\n# Starting ...\n');
   }
 
-  onTestResult (contexts, {testResults}) {
-    const text = [];
+  onTestResult (contexts, suite) {
+    const {testResults, testFilePath} = suite;
+
+    if (testFilePath) {
+      const filePathParts = testFilePath.split('/');
```
Comment (streamich) [inline src/TapReporter.js]:
```js
const filePathParts = testFilePath.split(require('fs').sep);
```

File: src/TapReporter.js [hunk 2]
```diff
@@ -1,60 +1,85 @@
...
+      const filePath = filePathParts.slice(0, filePathParts.length - 2).join('/');
```
Comment (streamich) [inline src/TapReporter.js]:
```js
const filePath = filePathParts.slice(0, filePathParts.length - 2).join(require('fs').sep);
```

File: src/TapReporter.js [hunk 3]
```diff
@@ -1,60 +1,85 @@
...
+      const fileName = filePathParts[filePathParts.length - 1];
+
+      this.logger.info(`\n${chalk.grey('#')}${chalk.bgBlue(' SUITE ')} ${chalk.grey(`${filePath}/`)}${fileName}`);
+    }
 
-    testResults.forEach((test, idx) => {
+    testResults.forEach((test) => {
       if (test.status === 'passed') {
```
Comment (streamich) [inline src/TapReporter.js]: Could we still have some kinda counter?

```js
this.logger.log(`${chalk.green('ok')} ${this.count++} ${test.title}`);
```

File: src/helpers/Logger.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  }
+
+  error (...args) {
+    if (this[level] >= LEVELS.ERROR) {
+      this.log(...args);
+    }
+  }
+
+  log (...args) {
+    this.log(...args);
```
Comment (streamich) [inline src/helpers/Logger.js]: Is this recursive?

Comment (carpasse) [reply src/helpers/Logger.js]: no and yes? this is working without an infinite loop not sure why 
will fix it.

File: src/TapReporter.js (hunk 3 above)
Comment (carpasse) [reply src/TapReporter.js]: Fiiiiiiine but if find it misleading

File: src/TapReporter.js [hunk 5]
```diff
@@ -1,60 +1,89 @@
...
 
-  onTestResult (contexts, {testResults}) {
-    const text = [];
+  onTestResult (contexts, suite) {
+    const {testResults, testFilePath} = suite;
+
+    if (testFilePath) {
+      const {dir, base} = path.parse(testFilePath);
+
+      this.logger.info(`\n${chalk.grey('#')}${chalk.bgBlue(' SUITE ')} ${chalk.grey(`${dir}${path.sep}`)}${base}`);
```
Comment (streamich) [inline src/TapReporter.js]: Maybe make the background of word `SUITE` red if the fails, and green if succeeds, like in Jest. And text color white.

Comment (carpasse) [reply src/TapReporter.js]: the file name is blank and the path is grey in Jest.
I will do the other change in a separate PR

**Pull request merged**